retrieval/manager.go

```diff
@@ -297,6 +302,12 @@ Outer:
 				produced++
 			}
 
+			if !r.leaderCandidate.IsLeader() {
+				common.SkippedPoints.Add(ctx, int64(len(outputs)), attribute.String("reason", "not_leader"))
```

Comment:
nit: there's a constant, `common.DroppedKeyReason` that is used for various kinds of drop already, maybe use that for consistency. It's named `key_reason` (twice in this method). Then maybe also rename `common.DroppedKeyReason` to just `common.KeyReason`.

Reply:
Yeah, that seems better, will do it!